### ISSUE TITLE
integrate numpy datetime64 for greater range of years

### ISSUE THREAD

**Issue by rlskoeser** [dh-tech/undate-python#77]
- how much does the code change? how much does numpy.datetime64 help, how heavy of a dependency is this?
- could this be optional, for people who need the expanded year range but not installed by default?

benchmarking / documentation:
- [x] how long does it take to run the test suite with/without numpy?
- [x] how big is the installation size with/without?

**Comment by rlskoeser** [comment]
next steps:
- check numpy depedencies; check 1.2 and 2.0 - difference in size? does the change break anything?
- fully implement clean numpy datetime shims and add typing, tests, docs, etc
- document numpy integration and any information about recommended versions, dependencies, etc.

**Comment by rlskoeser** [comment]
From NumPy installation docs:

> NumPy doesn’t depend on any other Python packages, however, it does depend on an accelerated linear algebra library - typically [Intel MKL](https://software.intel.com/en-us/mkl) or [OpenBLAS](https://www.openblas.net/). Users don’t have to worry about installing those (they’re automatically included in all NumPy install methods)

**Comment by rlskoeser** [comment]
@ColeDCrawford as I close this out, I remembered you wanted some benchmarking / assessment on the numpy inclusion.  The test suite is pretty fast either way, and we've also been expanding the functionality in other ways, so it's not a strict comparison.

Running a checkout of the 0.2 tag, I get
> 113 passed in 0.11s

Running in current develop, I get
> 138 passed in 0.16s

I compared the virtualenv size when installing version 0.2 and current develop. Without numpy, the virtualenv is 14MB; with numpy it's 46MB. Looking at the folders in the sitep ackages for the latter, the numpy folder is 31MB on its own.

Let me know if you think any of this needs to be documented in the changelog or elsewhere.
